### PR TITLE
feat(systemd-repart): introducing systemd-repart module

### PR DESCRIPTION
**Pull request by johannbg**
This modules introduces systemd-repart which can be used
to generate partition(s), via a simple systemd-repart
drop-in(s), which can be useful in wide varity of usecase.

This module is a standalone systemd based module as in it requires 
no other module than systemd.

## Checklist
- [ ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #

### CODE REVIEW

**Review by danimo — APPROVED** [review]

File: modules.d/01systemd-repart/module-setup.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+check() {
+
+    # If the binary(s) requirements are not fulfilled
+    # return 1 to not include the binary.
+    require_binaries systemd-repart || return 1
+
+    # If the module dependency requirements are not fulfilled
+    # return 1 to not include the required module(s).
+    if ! dracut_module_included "systemd"; then
+        derror "systemd-coredump needs systemd in the initramfs."
```
Comment (haraldh) [inline modules.d/01systemd-repart/module-setup.sh:15]: this is probably the wrong error message

**Review by haraldh — CHANGES_REQUESTED** [review]

File: modules.d/01systemd-repart/module-setup.sh (hunk 1 above)
Comment (johannbg) [reply modules.d/01systemd-repart/module-setup.sh]: Fixed.

**Comment by danimo** [comment]
@haraldh second round pls.

**Review by haraldh — APPROVED** [review]

File: modules.d/01systemd-repart/module-setup.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    # Install the required file(s.
+    inst_multiple -o \
+        # Install vendor repartition configurations
+        $libdir/repart.d/*.conf
+        # Install the systemd type service unit for systemd repart.
+        $systemdsystemunitdir/systemd-repart.service \
+        $systemdsystemunitdir/initrd-root-fs.target.wants/systemd-repart.service
+        # Install the binary executable(s) for systemd repart.
+        systemd-repart
```
Comment (haraldh) [inline modules.d/01systemd-repart/module-setup.sh:45]: Doesn't work with comments breaking line continuation... oversight

**Comment by haraldh** [comment]
So, we should at least test:
```
$ ./dracut.sh -l -a systemd-repart --no-kernel -f test-dracut.img
```

**Comment by haraldh** [comment]
which fails with:
```
dracut: *** Including module: systemd-repart ***
dracut-install: No SOURCE argument given
Usage: dracut-install -D DESTROOTDIR [-r SYSROOTDIR] [OPTION]... -a SOURCE...
or: dracut-install -D DESTROOTDIR [-r SYSROOTDIR] [OPTION]... SOURCE DEST
or: dracut-install -D DESTROOTDIR [-r SYSROOTDIR] [OPTION]... -m KERNELMODULE [KERNELMODULE …]

Install SOURCE (from rootfs or SYSROOTDIR) to DEST in DESTROOTDIR with all needed dependencies.

  KERNELMODULE can have the format:
     <absolute path> with a leading /
     =<kernel subdir>[/<kernel subdir>…] like '=drivers/hid'
     <module name>

  -D --destrootdir  Install all files to DESTROOTDIR as the root
  -r --sysrootdir   Install all files from SYSROOTDIR
  -a --all          Install all SOURCE arguments to DESTROOTDIR
  -o --optional     If SOURCE does not exist, do not fail
  -d --dir          SOURCE is a directory
  -l --ldd          Also install shebang executables and libraries
  -L --logdir <DIR> Log files, which were installed from the host to <DIR>
  -R --resolvelazy  Only install shebang executables and libraries
                     for all SOURCE files
  -H --hostonly     Mark all SOURCE files as hostonly

  -f --fips         Also install all '.SOURCE.hmac' files

  --module,-m       Install kernel modules, instead of files
  --kerneldir       Specify the kernel module directory
  --firmwaredirs    Specify the firmware directory search path with : separation
  --silent          Don't display error messages for kernel module install
  --modalias        Only generate module list from /sys/devices modalias list
  -o --optional     If kernel module does not exist, do not fail
  -p --mod-filter-path      Filter kernel modules by path regexp
  -P --mod-filter-nopath    Exclude kernel modules by path regexp
  -s --mod-filter-symbol    Filter kernel modules by symbol regexp
  -S --mod-filter-nosymbol  Exclude kernel modules by symbol regexp
  -N --mod-filter-noname    Exclude kernel modules by name regexp

  -v --verbose      Show more output
     --debug        Show debug output
     --version      Show package version
  -h --help         Show this help

dracut: FAILED: /home/harald/git/dracut/dracut-install -D /var/tmp/dracut.y4EItx/initramfs -a -o
/home/harald/git/dracut/modules.d/01systemd-repart/module-setup.sh: line 40: /repart.d/*.conf: No such file or directory
/home/harald/git/dracut/modules.d/01systemd-repart/module-setup.sh: line 43: /usr/lib/systemd/system/systemd-repart.service: Permission denied

```

**Pull request merged**